src/utils/propUtils.js

```diff
@@ -60,3 +63,15 @@ export const customPropTypes = {
   },
 }
 
+// ------------------------------------
+// Prop Renderers
+// ------------------------------------
+export const iconPropRenderer = (val) => _.isString(val) ? <Icon className={val} /> : val
+export const imagePropRenderer = (val) => _.isString(val) ? <Image src={val} /> : val
```

Comment:
Props that should behave consistently across all components will have an associated render util.  This allows us to standardize prop functionality and features.

In this case, `icon` and `image` props can render either a string or a component.